topological_action_planner/scripts/tap.py

```diff
@@ -0,0 +1,149 @@
+            )  # Based on where they are in ED?
+
+        if req.origin.entity == "":
+            # This indicates the plan starts from the robot's current pose
+            # We'll insert a node corresponding to our current position.
+            # Difficulty is determining it's adjacency to other nodes, based on geometry
+            robot_name = rospy.get_namespace().split("/")[-2]
+            robot_entity = self._wm.get_entity(robot_name)
+            origin_node = robot_name, ""
+            # Ask ED in which room the robot is currently, maybe based on it's pose
+            current_rooms = rooms_of_entity(self._wm, robot_entity)
+            current_room_ids = [r.uuid for r in current_rooms]
+            # TODO: This assumes that the robot can always drive to any other node in the same room.
+            # This might not always be true of course. It may also make more sense to only connect with the closest N waypoints
+            nodes_in_same_room = [
+                node for node, info in graph.nodes.items() if info["room"][0].uuid in current_room_ids
+            ]
+            if not nodes_in_same_room:
+                # TODO: What happens when the robot is not in a defined room? But eg. just outside the apartment
+                rospy.logerr("There are no nodes to connect the robot to")
+            for node in nodes_in_same_room:
+                # TODO: Will be updated later when taking path length into account later
+                graph.add_edge(origin_node, node, action_type=Edge.ACTION_DRIVE, weight=1)
+        else:
+            origin_node = req.origin.entity, req.origin.area
+
+        dst_node = req.destination.entity, req.destination.area
+        rospy.loginfo("Requesting topological action plan from {} to {}".format(origin_node, dst_node))
+
+        try:
+            edges = self._topo_planner.plan(graph, origin_node, dst_node)
+
+            self._pub_grasp_marker.publish(create_tap_marker_array(graph, self._wm))
+            rospy.loginfo("Found plan of {} edges".format(len(edges)))
+            return PlanResponse(error_msg="", error_code=PlanResponse.SUCCESS, edges=edges)
+        except nx.NetworkXNoPath as no_path_found_ex:
+            rospy.logerr(no_path_found_ex)
+            return PlanResponse(
+                error_msg=str(no_path_found_ex),
+                error_code=PlanResponse.ERROR_NO_PATH_FOUND,
+            )
+        except nx.NodeNotFound as node_not_found_ex:
+            rospy.logerr(node_not_found_ex)
+            return PlanResponse(
+                error_msg=str(node_not_found_ex),
+                error_code=PlanResponse.ERROR_UNKNOWN_NODE,
+            )
+
+    def _srv_update_edge_cb(self, req: UpdateEdgeRequest) -> UpdateEdgeResponse:
+        try:
+            self.update_edge(req.updated)
+            return UpdateEdgeResponse(success=True)
+        except nx.NodeNotFound as node_not_found_ex:
+            rospy.logerr(node_not_found_ex)
```

Comment:
```suggestion
        except nx.NodeNotFound as ex:
            rospy.logerr(ex)
```

Reply:
Do not agree, there's multiple exception types, I want to be clear about their differences.

Reply:
That is clear from the `except XXX` imo.